pkg/sharing/upload.go

```diff
@@ -95,11 +98,63 @@ func (s *Sharing) UploadTo(inst *instance.Instance, m *Member) (bool, error) {
 	}
 	inst.Logger().WithField("nspace", "upload").Debugf("lastSeq = %s", lastSeq)
 
-	seq := lastSeq // TODO WIP
+	file, seq, err := s.findNextFileToUpload(inst, lastSeq)
+	if err != nil {
+		return false, err
+	}
+	if file == nil {
+		if seq != lastSeq {
+			err = s.UpdateLastSequenceNumber(inst, m, "upload", seq)
+		}
+		return false, err
+	}
 
-	if seq == lastSeq {
-		return false, nil
+	if err = s.uploadFile(inst, m, file); err != nil {
+		return false, err
 	}
 
 	return true, s.UpdateLastSequenceNumber(inst, m, "upload", seq)
 }
+
+// findNextFileToUpload uses the changes feed to find the next file that needs
+// to be uploaded. It returns a file document if there is one file to upload,
+// and the sequence number where it is in the changes feed.
+func (s *Sharing) findNextFileToUpload(inst *instance.Instance, since string) (*vfs.FileDoc, string, error) {
+	for {
```

Comment:
Isn't a bit dangerous? Maybe a MAX_CHANGES condition could be added

Reply:
Well, maybe we can